client/src/components/SentimentAnalysis/SentimentAnalysis.js

```diff
@@ -1,45 +1,86 @@
 import tardigrade from '../../images/tardigrade.png';
 
-const SentimentAnalysis = () => {
-  const [isVerified, setIsVerified] = useState(false);
-  const [header, setHeader] = useState({
-    entities: [],
-    sentiment: { score: 0.0, magnitude: 0.0 },
-  });
+const SentimentAnalysis = props => {
+  const locationState = props.location.state;
+  const ad = locationState !== undefined ? locationState.ad : DEFAULT_ANALYSIS;
 
-  const [content, setContent] = useState({
-    entities: [],
-    sentiment: { score: 0.0, magnitude: 0.0 },
-  });
+  const [isVerified, setIsVerified] = useState(false);
+  const [headline, setHeadline] = useState(ad.headlineAnalysis);
+  const [content, setContent] = useState(ad.contentAnalysis);
 
+  /**
+   * Creates an Ad Analysis object and passes it to
+   * the /analysis route (SentimentAnalysis component)
+   * @param {Boolean} value - whether the user has been
+   * verified by the capcha
+   * @returns {Void} doesn't return anything
+   */
   const onChange = value => {
     setIsVerified(value !== null);
   };
 
+  /**
+   * Set the current headline and content analysis of
+   * the component with new data
+   * @param {Boolean} data - new data returned from API
+   * @returns {Void} doesn't return anything
+   */
+  const updatePage = data => {
+    setHeadline({
+      entities: data.headline_entities,
+      sentiment: JSON.parse(data.headline_sentiment),
+    });
+    setContent({
+      entities: data.content_entities,
+      sentiment: JSON.parse(data.content_sentiment),
+    });
+  };
+
+  /**
+   * Send user-input form to API and get back data
+   * @param {FormSubmitEvent} event - event that targets
+   * the whole form
+   * @returns {Void} doesn't return anything
+   */
   const submitForm = event => {
     event.preventDefault();
     const form = new FormData(event.target);
-    const url = 'https://analysis-dot-step9-2020-capstone.appspot.com/analysis';
-
-    fetch(url, { method: 'POST', body: form })
+    fetch(API_URL, { method: 'POST', body: form })
       .then(response => response.json())
-      .then(data => {
-        const headerEntities = data.header_entities;
-        const headerSentiment = JSON.parse(data.header_sentiment);
-        setHeader({ entities: headerEntities, sentiment: headerSentiment });
+      .then(data => updatePage(data));
+  };
 
-        const contentEntities = data.content_entities;
-        const contentSentiment = JSON.parse(data.content_sentiment);
-        setContent({ entities: contentEntities, sentiment: contentSentiment });
-      });
+  /**
+   * Pick the appropriate value for textarea based on
+   * the given label
+   * @param {String} label - the label of textarea
+   * @returns {String} the appropriate value for textarea
+   * - If there's params passed from Search route, display those values
+   * - If not, show empty textarea so user can filled in themselves
+   */
+  const displayTextareaValue = label => {
+    if (locationState !== undefined) {
```

Comment:
Rewrite this to avoid nested if/elses:

if (locationState === undefined) {
  return '';
}

return label === 'headline' ? locationState.ad.headline : locationState.ad.content

Reply:
Ahhh I totally forgot about the ternary operator - thanks for the suggestion, the code looks so much better now :D